(API) Oracle Eloqua destination

mohan-h opened the pull request:
Initial commit

mohan-h committed:
(API) Oracle Eloqua destination

initial commit

mohan-h committed:
Merge remote-tracking branch 'upstream/main' into oracle-eloqua-api-destination

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+---
+keywords: crm;CRM;crm destinations;api oracle eloqua;api oracle eloqua crm destination
```
As per the documentation self-service template, we don't need to add keywords to the metadata.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+keywords: crm;CRM;crm destinations;api oracle eloqua;api oracle eloqua crm destination
+title: (API) Oracle Eloqua connection
+description: The (API) Oracle Eloqua destination allows you to export your account data and activate it within Oracle Eloqua for your business needs.
+exl-id: 
```
As per the documentation self-service template, please remove the `exl-id` line. The `exl-id` will be automatically generated after we publish the page. Including this line can actually adversely affect the search experience on the site.

mohan-h replied:
Removed however please check the [template](https://github.com/AdobeDocs/experience-platform.en/blob/main/help/destinations/destination-sdk/docs-framework/self-service-template.md) still has the exl-id.

vgiurgiu replied:
Ah, I see what happened here. The link you provide above is the Got location for the published example page on Experience League. Since this is a live page, it will have an `exl-id`. 

For the latest template, please always go to this link (https://experienceleague.adobe.com/docs/experience-platform/destinations/destination-sdk/document-destination/documentation-instructions.html?lang=en#documentation-template) and select **Download the template here** to get the latest version of the template, which does not have an `exl-id`.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+Before activating data to the [!DNL Oracle Eloqua] destination, you must have a [schema](/help/xdm/schema/composition.md), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segments/create-segments.html?lang=en) created in [!DNL E ...
+
+Refer to Adobe's documentation for [Segment Membership Details schema field group](/help/xdm/field-groups/profile/segmentation.md) if you need guidance on segment statuses.
```
One comments here: 

Let's please not use Adobe's documentation. Let's use Experience Platform instead, as in: "Refer to the Experience Platform documentation"

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+Note down the items below before you authenticate to the [!DNL Oracle Eloqua] destination:
+
+| Credential | Description |
+| --- | --- | --- |
```
Please revisit this table, as it renders as broken. The issue is most likely on line 38. 
Please also use a markdown previewer when you author the documentation, to catch these issues before submitting for review. Thanks!

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+>[!NOTE]
+>
+>* When creating contact fields used to save segment mappings, ensure the count falls within the maximum limit of 250 custom contact fields. 
```
When creating contact fields -> is this in Eloqua? please call this out clearly. 

Also, we are introducing this concept directly as a limitation -- what are contact fields and why are these needed to save segment mappings?

mohan-h replied:
Have reworded.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+>[!NOTE]
+>
+>* When creating contact fields used to save segment mappings, ensure the count falls within the maximum limit of 250 custom contact fields. 
+>* If the limit is exceeded you will encounter a `400: "There was a validation error"` error.
```
Do you encounter this error in Experience Platform or in Eloqua?

Also, a comma is needed before "you".

mohan-h replied:
Eloqua, have reworded. 
Thanks for the comma catch.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+>
+>* When creating contact fields used to save segment mappings, ensure the count falls within the maximum limit of 250 custom contact fields. 
+>* If the limit is exceeded you will encounter a `400: "There was a validation error"` error.
+>* To be able to update new segments remove some existing mappings from your datasets and delete the corresponding custom contact fields in your [!DNL Oracle Eloqua] account.
```
Please simplify this sentence, something like:

If you have reached the limit specified above, you need to remove existing mappings from your datasets (by the way, why are we talking about datasets??) and delete the corresponding custom contact fields in your [!DNL Oracle Eloqua] account before you can export more segments.

mohan-h replied:
Reworded.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+|Target Identity|Example|Description|Considerations|
+|---|---|---|---|
+| `EloquaId` | `111111` | Unique identifier of the contact.| **Mandatory**. |
```
No period needed after Mandatory

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+1. In the **[!UICONTROL Select source field]** window, choose the **[!UICONTROL Select attributes]** category and select the XDM attribute or choose the **[!UICONTROL Select identity namespace]** and select an identity.
+1. In the **[!UICONTROL Select target field]** window, choose the **[!UICONTROL Select identity namespace]** and select an identity or choose **[!UICONTROL Select custom attributes]** category and select an attribute as needed.
+    * Repeat these steps to add the following mappings between your XDM profile schema and your [!DNL Oracle Eloqua] instance:
+        |Source Field|Target Field| Mandatory|
```
Some clarification questions: 

1. Are we absolutely sure that **both** these fields (`Attribute: emailAddress` and `Identity: EloquaId`) are absolutely necessary? What happens if a user doesn't select one or both of them?
2. Do the mappings need to be exactly as described in the table? Namely, on the Source Field side, can users select other attributes/identities and map them to the Target fields?

mohan-h replied:
_Are we absolutely sure that both these fields (Attribute: emailAddress and Identity: EloquaId) are absolutely necessary? What happens if a user doesn't select one or both of them?_

Dev team : Yes ,EloquaID is needed in the http call as corresponding to that all other attributes are updated, Email Id is also necessary as without it the API throws error.

```
    {
        "type": "ObjectValidationError",
        "container": {
            "type": "ObjectKey",
            "objectType": "Contact"
        },
        "property": "emailAddress",
        "requirement": {
            "type": "EmailAddressRequirement"
        },
        "value": "<null>"
    }
]  
```

_Do the mappings need to be exactly as described in the table? Namely, on the Source Field side, can users select other attributes/identities and map them to the Target fields?_

No, All we need to consider is the target mapping. The attributes need to be named as described in table because these attributes will form request body.
From source side we don't have any restriction to map fields ,you can map from any where you want if the data is appropriate it will be pushed to destination else will throw an error.
(Example - you can map EloquaID to any identity namespace like contact key,outreach ID etc. but if that ID is not present in Eloqua DB it will throw an error.

vgiurgiu replied:
These are great clarifications, thanks!

Could we transform these clarifications into material that we could add to this documentation page?

mohan-h replied:
Sure, incorporated.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+>[!NOTE]
+>
+>The destination automatically suffixes a unique identifier to the selected segment names on each execution when sending the contact field information to [!DNL Oracle Eloqua]. This ensures the contact field names corresponding to your segment names do not overlap.
```
Can we please provide an example here of what the exported segment with a suffix looks like? or link to somewhere else in the doc where we show an example.

mohan-h replied:
Reworded and added reference to the validation section.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count within the segment.
+![Platform UI screenshot example showing Segment.](../../assets/catalog/email-marketing/oracle-eloqua-api/segment.png)
+
+1. Log in to the [!DNL Oracle Eloqua] website, then navigate to the [!DNL Contacts Overview] page to check if the profiles from the segment have been added. To see the segment status drill down into a [!DNL Contact Detail] page and check if the contact field having the selected segment name as its prefix has been created.
```
UI elements should be in bold, so that they stand out in the text. this includes UI elements in Eloqua, like [!DNL Contacts Overview] and [!DNL Contact Detail].

minor grammar suggestions: 

To see the segment status, drill down into a [!DNL Contact Detail] page and check if the contact field with the selected segment name as its prefix has been created.

mohan-h replied:
Thanks

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+## Errors and troubleshooting {#errors-and-troubleshooting}
+
+If the [!DNL Oracle Eloqua] API fails to validate a request, it will respond with a `400: "There was a validation error"` error message describing the issue.
```
Minor, but we try to use present tense over future tense. for example: 

If the [!DNL Oracle Eloqua] API fails to validate a request, it responds with a `400: "There was a validation error"` error message describing the issue.

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,165 @@
+
+If the [!DNL Oracle Eloqua] API fails to validate a request, it will respond with a `400: "There was a validation error"` error message describing the issue.
+
+First check by ensuring you have met the conditions defined in the (guardrails)[#guardrails] section.
```
As a first step, make sure you have met the conditions outlined in the [guardrails](#guardrails) section.

mohan-h replied:
Updated.

vgiurgiu reviewed (commented):
Good work! I left some comments, please let us know if you have any questions. Thanks!

mohan-h committed:
Merge remote-tracking branch 'upstream/main' into oracle-eloqua-api-destination

mohan-h committed:
Updates against review

Updates post review https://github.com/AdobeDocs/experience-platform.en/pull/151#pullrequestreview-1199660687

mohan-h committed:
Removed extraneous heading

mohan-h committed:
Removed the Export type and frequency section

Removed the Export type and frequency section as its not used in (API) Oracle ELoqua

vgiurgiu commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,154 @@
+>[!NOTE]
+>
+>* [!DNL Oracle Eloqua] has a maximum limit of 250 custom contact fields.
+>* [!DNL Oracle Eloqua] custom contact fields are created using the names of the segments selected during the **[!UICONTROL Destinations]** > **[!UICONTROL Activate Destination]** > **[!UICONTROL Select segments]** step.
```
Thanks for updating this part. I just have a follow-up question. Are [!DNL Oracle Eloqua] custom contact fields created automatically in Eloqua or does the user need to create these? 

If these are created automatically, my suggestion is to add this callout to make things as clear as possible.

mohan-h replied:
They are created automatically, have reworded.

vgiurgiu reviewed (commented):
Thanks for the updates, I left two more comments, looking for some minor clarifications.

mohan-h committed:
Content updates

mohan-h committed:
Content updates

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+[[!DNL Oracle Eloqua]](https://www.oracle.com/cx/marketing/automation/) enables marketers to plan and execute campaigns while delivering a personalized customer experience for their prospects. With integrated lead management and easy campaign creation, it helps marketers engage the right audience at the right time in their buyer's journey and elegantly scales to reach audiences across channels in ...
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Oracle Eloqua] Update a Contact REST API](https://docs.oracle.com/en/cloud/saas/marketing/eloqua-rest-api/op-api-rest-1.0-data-contact-id-put.html), which allows you to update identities within a segment into [!DNL Oracle Eloqua].
```
"Update a contact" is just an operation supported by their REST API, it's not an API itself.
 Let's rephrase that to "[...] leverages the [Update a contact](https://docs.oracle.com/en/cloud/saas/marketing/eloqua-rest-api/op-api-rest-1.0-data-contact-id-put.html) operation from the Oracle Eloqua REST API [...]"

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Oracle Eloqua] Update a Contact REST API](https://docs.oracle.com/en/cloud/saas/marketing/eloqua-rest-api/op-api-rest-1.0-data-contact-id-put.html), which allows you to update identities within a segment into [!DNL Oracle Eloqua].
+
+[!DNL Oracle Eloqua] uses [Basic Authentication](https://docs.oracle.com/en/cloud/saas/marketing/eloqua-rest-api/Authentication_Basic.html) to communicate with the [!DNL Oracle Eloqua Update a Contact REST API]. Instructions to authenticate to your [!DNL Oracle Eloqua] instance are further below, in the [Authenticate to destination](#authenticate) section.
```
"to communicate with the [!DNL Oracle Eloqua REST API]"

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+| Credential | Description |
+| --- | --- |
+| `Username` | The Username of your [!DNL Oracle Eloqua] account. |
```
Username and password in the Description column should be lowercase.

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+>[!NOTE]
+>
+>* [!DNL Oracle Eloqua] custom contact fields are automatically created using the names of the segments selected during the **[!UICONTROL Destinations]** > **[!UICONTROL Activate Destination]** > **[!UICONTROL Select segments]** step.
```
No need to specify the entire path to a step when referring to said step. Please rephrase as "[!DNL Oracle Eloqua] custom contact fields are automatically created using the names of the segments selected during the  **[!UICONTROL Select segments]** step."

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+>* [!DNL Oracle Eloqua] custom contact fields are automatically created using the names of the segments selected during the **[!UICONTROL Destinations]** > **[!UICONTROL Activate Destination]** > **[!UICONTROL Select segments]** step.
+
+* [!DNL Oracle Eloqua] has a maximum limit of 250 custom contact fields.
+* Before exporting new segments ensure you are within this limit by logging into your [!DNL Oracle Eloqua] account.
```
This part is confusing. You're saying that Eloqua has a maximum limit of 250 custom contact fields. Then you ask readers to make sure they are within the limit, by checking their Eloqua account. I don't see how this makes sense. If the limit is on Eloqua's side, you will always be within that limit. That's what a limit does, right?

I believe what you meant here is to ask readers to check their Platform account and make sure that the number of segments they are exporting + the number of existing segments in Eloqua do not exceed the limit. We should be very clear on this behavior.

mohan-h replied:
Thanks, have rephrased

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+### Authenticate to destination {#authenticate}
+
+Fill in the required fields below. Refer to the [Gather [!DNL Oracle Eloqua] credentials](#gather-credentials) section for any guidance.
+* **[!UICONTROL Password]**: The Password of your [!DNL Oracle Eloqua] account.
```
Password and username in lowercase in the description.

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+>[!IMPORTANT]
+>
+>Both the attribute mappings listed in the table which follows are mandatory.
```
Which table? There is no table here.

mohan-h replied:
Changed the position to just below the table.

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+>[!NOTE]
+>
+>The destination automatically suffixes a unique identifier to the selected segment names on each execution when sending the contact field information to [!DNL Oracle Eloqua]. This ensures the contact field names corresponding to your segment names do not overlap. Refer to the [Validate data export](#exported-data) section screenshot example of a [!DNL Oracle Eloqua] Contact Details page with cus ...
```
"**an** [!DNL Oracle Eloqua] Contact Details page"

vfalon commented on help/destinations/catalog/email-marketing/oracle-eloqua-api.md:
```diff
@@ -0,0 +1,183 @@
+
+## Errors and troubleshooting {#errors-and-troubleshooting}
+
+When creating the destination, if you obtain an error message: `400: There was a validation error` or `400 BAD_REQUEST` as the [!DNL Oracle Eloqua] API response.
```
The phrasing here sounds incomplete/incorrect. Let's rephrase this.

"When creating the destination, you might receive one of the following error messages: `400: There was a validation error` or `400 BAD_REQUEST`. This happens when you exceed the 250 custom contact fields limit, as described in the [guardrails](#guardrails) section. To fix this error, make sure you are not exceeding the limit.

vfalon reviewed (changes requested):
I've left a few comments to address. Let me know if you have any questions.

mohan-h committed:
Updated post review
